src/Digest.ts

```diff
@@ -27,14 +27,15 @@ const lengthAndInput = (input: Uint8Array): Uint8Array => u8a.concat([writeUint3
 // This implementation of concatKDF was inspired by these two implementations:
 // https://github.com/digitalbazaar/minimal-cipher/blob/master/algorithms/ecdhkdf.js
 // https://github.com/panva/jose/blob/master/lib/jwa/ecdh/derive.js
-export function concatKDF(secret: Uint8Array, keyLen: number, alg: string): Uint8Array {
+export function concatKDF(secret: Uint8Array, keyLen: number, alg: string, apu?:string, apv?:string): Uint8Array {
   if (keyLen !== 256) throw new Error(`Unsupported key length: ${keyLen}`)
   const value = u8a.concat([
     lengthAndInput(u8a.fromString(alg)),
-    lengthAndInput(new Uint8Array(0)), // apu
-    lengthAndInput(new Uint8Array(0)), // apv
+    lengthAndInput((apu == undefined) ? new Uint8Array(0) : u8a.fromString(apu)),
+    lengthAndInput((apv == undefined) ? new Uint8Array(0) : u8a.fromString(apv)),
```

Comment:
This is supposed to be `utf8` to bytes?

Reply:
I will change the parameters for apu and apv to producer/consumerInfo as UInt8Arrays. 

The JWE spec defines what values and encodings are expected. In case of 1PU, apu and apv supposed to be the base64url decoded apu/apv which in 1PU's case is always some UTF-8 bytes but to be more general, I will change that.